tests/foreman/ui_airgun/test_ldap_authentication.py

```diff
@@ -241,3 +311,343 @@ def test_positive_create_with_idm_org_and_loc(session, ipa_data):
+            {
+                'resource_type': 'Activation Keys',
+                'permission.assigned': PERMISSIONS_UI['Activation Keys'],
+            }
+        )
+        session.user.update(
+            ldap_data['ldap_user_name'], {'roles.resources.assigned': [katello_role]})
+    with Session(
+            test_name,
+            ldap_data['ldap_user_name'],
+            ldap_data['ldap_user_passwd'],
+    ) as session:
+        if bz_bug_is_open(1652938):
+            try:
+                session.activationkey.search('')
+            except NoSuchElementException:
+                session.browser.refresh()
+        session.activationkey.create({'name': ak_name})
+        assert session.activationkey.search(ak_name)[0]['Name'] == ak_name
+        current_user = session.activationkey.read(ak_name)['current_user']
+        assert current_user == ldap_data['ldap_user_name']
+
+
+@tier2
+def test_positive_add_admin_role_with_org_loc(
+        session, ldap_data, ldap_user_name, test_name, auth_source, ldap_usergroup_name,
+        module_org):
+    """Associate Admin role to User Group with org and loc set.
+    [belonging to external AD User Group.]
+
+    :id: 00841778-f89e-4445-a6c6-f1470b6da32e
+
+    :setup: LDAP Auth Source should be created with Org and Location
+            Associated.
+
+    :Steps:
+        1. Create an UserGroup.
+        2. Assign admin role to UserGroup.
+        3. Create and associate an External AD UserGroup.
+
+    :expectedresults: Whether a User belonging to User Group is able to
+        access some of the pages, with the associated org and loc
+        in LDAP Auth source page as the context set.
+
+    :CaseImportance: Critical
+    """
+    ak_name = gen_string('alpha')
+    location_name = gen_string('alpha')
+    with session:
+        session.usergroup.create({
+            'usergroup.name': ldap_usergroup_name,
+            'roles.admin': True,
+            'external_groups.name': 'foobargroup',
+            'external_groups.auth_source': 'LDAP-' + auth_source.name,
+        })
+        assert session.usergroup.search(ldap_usergroup_name)[0]['Name'] == ldap_usergroup_name
+        session.user.update(ldap_data['ldap_user_name'], {'user.auth': 'LDAP-' + auth_source.name})
+    with Session(
+            test_name,
+            ldap_data['ldap_user_name'],
+            ldap_data['ldap_user_passwd'],
+    ) as session:
+        session.location.create({'name': location_name})
+        assert session.location.search(location_name)[0]['Name'] == location_name
+        location = session.location.read(location_name)
+        assert location['current_user'] == ldap_data['ldap_user_name']
+        assert location['primary']['name'] == location_name
+        if bz_bug_is_open(1652938):
+            try:
+                session.activationkey.search('')
+            except NoSuchElementException:
+                session.browser.refresh()
+        session.organization.select(module_org.name)
```

Comment:
nitpick: select both module_org and module_loc and operate with any entity in that context

Reply:
I'd rather not - at this point location `location_name` is selected. Since user is admin it doesn't really matter to him in which context to operate with entities, but that `location_name` loc is created for this test specifically, so such scenario is a bit more isolated and there's less chance to affect other tests.
I can change to `module_loc` if you still think that's the better way, but i can't see any win in that.